frontend/storage/utils/downloadGameData.ts

```diff
@@ -0,0 +1,53 @@
+import { AnyAction, Dispatch } from 'redux';
+import { SERVER_ADRESS, DataLoadActions, BreadcrumbActions } from '../common';
+import axios, { AxiosResponse } from 'axios';
+import { GameSeries } from '../models/GameSeries';
+import { Game } from '../models/Game';
+import { Image } from '../models/Image';
+import { GameEnvironment } from '../models/GameEnvironment';
+import { Environment } from '../models/Environment';
+
+export async function requestGameData(id: number, dispatch: Dispatch<AnyAction>, changeGameExistsState: (doesGameExistsInDb: boolean) => void) {
```

Comment:
The reason the code in this function is so complex is simple: it does too many things at once. You have one function that does the following:
1. Request game from API
2. Handle failure state of the above
3. If game was loaded then load the connected series and also all environment-links
4. Handle failure state of the above
5. Load all the environments
6 Handle failure state of the above
7. Put everything into the store

Ooof, that's a lot! You could separate this into a few layers:
1. A class responsible for making the actual API requests (essentially wrap `axios` with your own layer of abstraction). This will simplify all the requests. And will allow you to easily handle `error` in the API responses.
2. Have a separate function for fetching each model from the database, which could also automatically put loaded models into the store.
3. Then, you could have a function that loads all of the necessary data to show a game which would be something like:


```
function loadAllNecessaryDataToShow_a_game(gameId) {
    try {
    const game = await loadGame(gameId);
    const gameEnvironments = await loadGameEnvironmentsByGame(gameId);
    const environments = await loadEnvironments(gameEnvironments.map(x => x.id));
    const serie = await loadGameSerie(game.seriesId);
        return game;
    } catch (e){
        return undefined;
    }
}
```